cvmfs/cvmfs.cc

```diff
@@ -1332,9 +1339,8 @@ static void cvmfs_open(fuse_req_t req, fuse_ino_t ino,
   // fd < 0
   // the download has failed. Close the page cache tracker entry if required
   if (!dirent.IsDirectIo()) {
-    fuse_remounter_->fence()->Enter();
+    assert(!open_directives.direct_io);
```

Comment:
Test 708 seems to trigger this assert

```
Thu Oct 31 10:43:57 2024 (test.cern.ch) CernVM-FS: running in debug mode
Thu Oct 31 10:43:57 2024 (test.cern.ch) Warning: current limits for number of open files are (1024/1048576)
CernVM-FS is likely to run out of file descriptors, set ulimit -n to at least 8192
Thu Oct 31 10:43:57 2024 (test.cern.ch) re-building cache database
Thu Oct 31 10:43:57 2024 (test.cern.ch) (manager 'standard') switching proxy from (none) to DIRECT. Reason: set random start proxy from the first proxy group [Current host: http://localhost/cvmfs/test.cern.ch]
Thu Oct 31 10:43:57 2024 (test.cern.ch) (manager 'external') switching proxy from (none) to DIRECT. Reason: cloned [Current host: http://localhost/cvmfs/test.cern.ch]
Thu Oct 31 10:43:57 2024 (test.cern.ch) CernVM-FS: linking /home/vavolkl/r/cvmfs/test/src/708-pagecachecornercase/mnt to repository test.cern.ch
Thu Oct 31 10:43:58 2024 (test.cern.ch) switched to catalog revision 4
Thu Oct 31 10:44:00 2024 (test.cern.ch) (manager 'standard') failed to resolve IP addresses for no-such-host.cern.ch (4 - unknown host name)
Thu Oct 31 10:44:00 2024 (test.cern.ch) (manager 'standard') switching proxy from DIRECT to http://no-such-host.cern.ch:3128 (no-such-host.cern.ch). Reason: set random start proxy from the first proxy group [Current host: http://localhost/cvmfs/test.cern.ch]
Thu Oct 31 10:44:01 2024 (test.cern.ch) failed to fetch /foo (hash: c5baca6f42d65e4a21dc9bad66dea5152ab813d8, error 6 [proxy connection problem])
Thu Oct 31 10:44:03 2024 (test.cern.ch) --
Signal: 6, errno: 2, version: 2.12.0, PID: 17655
Executable path: /usr/bin/cvmfs2
failed to re-gain root permissions... still give it a try 

Thread 15 (Thread 0x7f2b3f7fe700 (LWP 17673)):
#0  0x00007f2b71db71eb in __GI___select (nfds=0, readfds=0x0, writefds=0x0, exceptfds=0x0, timeout=0x7f2b3f7fcb70) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x00007f2b713fb7a3 in SafeSleepMs (ms=100) at /home/vavolkl/r/cvmfs/cvmfs/util/posix.cc:2053
#2  0x00007f2b7174adf5 in Watchdog::SendTrace (sig=6, siginfo=0x7f2b3f7fcfb0, context=0x7f2b3f7fce80) at /home/vavolkl/r/cvmfs/cvmfs/monitor.cc:328
#3  <signal handler called>
#4  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#5  0x00007f2b71cc4859 in __GI_abort () at abort.c:79
#6  0x00007f2b71cc4729 in __assert_fail_base (fmt=0x7f2b71e5a588 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", assertion=0x7f2b719b3af5 "!open_directives.direct_io", file=0x7f2b719b32b8 "/home/vavolkl/r/cvmfs/cvmfs/cvmfs.cc", line=1342, function=<optimized out>) at assert.c:92
#7  0x00007f2b71cd5fd6 in __GI___assert_fail (assertion=0x7f2b719b3af5 "!open_directives.direct_io", file=0x7f2b719b32b8 "/home/vavolkl/r/cvmfs/cvmfs/cvmfs.cc", line=1342, function=0x7f2b719b3a70 "void cvmfs::cvmfs_open(fuse_req_t, fuse_ino_t, fuse_file_info*)") at assert.c:101
#8  0x00007f2b717e30e5 in cvmfs::cvmfs_open (req=0x7f2b2c001670, ino=257, fi=0x7f2b3f7fdd30) at /home/vavolkl/r/cvmfs/cvmfs/cvmfs.cc:1342
#9  0x00007f2b71b107d5 in loader::stub_open (req=0x7f2b2c001670, ino=257, fi=0x7f2b3f7fdd30) at /home/vavolkl/r/cvmfs/cvmfs/loader.cc:265
```

Reply:
Interesting. It doesn't on my machine... ok, checking.

Reply:
Mine is arch linux, and it does seem to work on FC40 but not on the other platforms. So I assume it has to do with older/newer kernel.